pkg/externalfunctions/dataextraction.go

```diff
@@ -0,0 +1,465 @@
+package externalfunctions
+
+import (
+	"bytes"
+	"context"
+	"crypto/sha256"
+	"encoding/hex"
+	"fmt"
+	"log"
+	"os"
+	"path/filepath"
+	"strings"
+	"sync"
+
+	"github.com/google/go-github/v56/github"
+	"github.com/google/uuid"
+	"github.com/pandodao/tokenizer-go"
+	"github.com/tmc/langchaingo/documentloaders"
+	"github.com/tmc/langchaingo/schema"
+	"github.com/tmc/langchaingo/textsplitter"
+)
+
+// DataExtractionGetGithubFilesToExtract gets all files from github that need to be extracted.
+//
+// Parameters:
+//   - githubRepoName: name of the github repository.
+//   - githubRepoOwner: owner of the github repository.
+//   - githubRepoBranch: branch of the github repository.
+//   - githubAccessToken: access token for github.
+//   - githubFileExtensions: github file extensions.
+//   - githubFilteredDirectories: github filtered directories.
+//   - githubExcludedDirectories: github excluded directories.
+//
+// Returns:
+//   - githubFilesToExtract: github files to extract.
+func DataExtractionGetGithubFilesToExtract(githubRepoName string, githubRepoOwner string,
+	githubRepoBranch string, githubAccessToken string, githubFileExtensions []string,
+	githubFilteredDirectories []string, githubExcludedDirectories []string) (githubFilesToExtract []string) {
+
+	client, ctx := dataExtractNewGithubClient(githubAccessToken)
+
+	// Retrieve the specified branch's SHA (commit hash) from the GitHub repository. This is used to identify the latest state of the branch.
```

Comment:
```suggestion
	// Retrieve the specified branch SHA (commit hash) from the GitHub repository. This is used to identify the latest state of the branch.
```

Try committing this change... fingers crossed